streams/src/test/java/org/apache/kafka/streams/processor/internals/RepartitionTopicsTest.java

```diff
@@ -168,12 +171,16 @@ public void shouldReturnMissingSourceTopics() {
             clusterMetadata,
             "[test] "
         );
+        repartitionTopics.setup();
 
-        assertThat(repartitionTopics.setup(), equalTo(false));
         assertThat(
-            repartitionTopics.missingUserInputTopicsPerTopology(),
-            equalTo(Collections.singletonMap(UNNAMED_TOPOLOGY, missingSourceTopics))
+            repartitionTopics.topologiesWithMissingInputTopics(),
+            equalTo(Collections.singleton(UNNAMED_TOPOLOGY))
         );
+        final StreamsException exception = repartitionTopics.missingSourceTopicExceptions().poll();
+        assertThat(exception, notNullValue());
+        assertThat(exception.taskId().isPresent(), is(true));
+        assertThat(exception.taskId().get(), equalTo(new TaskId(0, 0)));
```

Comment:
Could you verify that in the absence of missing input topics `topologiesWithMissingInputTopics()` and  `missingSourceTopicExceptions()` return empty collections?